src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -250,9 +250,12 @@ void updateCACertificateConfig(List<String> caCerts) {
 
     @Override
     protected CompletableFuture<Void> close(boolean waitForDependers) {
-        // shutdown the threadpool in close, not in shutdown() because it is created
-        // and injected in the constructor and we won't be able to restart it after it stops.
-        cloudCallThreadPool.shutdown();
+        if (cloudCallThreadPool != null) {
```

Comment:
I am adding this here because I saw `NullPointer` exceptions get thrown occasionally when close was being called before this had been initialized